### PR TITLE
- fix Windows -> Android cross-compilation (and probably others)

### LINKED ISSUE #28: [package] Error building OpenSSL for Android
While building OpenSSL for android on macOS I've seen the following issue when building for arm:

`openssl/conanfile.py` modifies `Makefile.org` with a string replace `AR=arm-linux-androideabi-ar` this replace is somehow triggered two times resulting in `AR=arm-linux-androideabi-arm-linux-androideabi-ar`.

I think the problem always triggers when using an `arm-` prefix as this also matches searching for `AR=ar`

My conanfile.txt:
```
[requires]
OpenSSL/1.0.2s@conan/stable
Poco/1.9.3@pocoproject/stable

[generators]
cmake
```

My profile:
```
ndk_root=$HOME/Library/Android/sdk/ndk-bundle
standalone_toolchain=$HOME/Development/toolchains/android-16-arm/
target_host=arm-linux-androideabi
cc_compiler=clang
cxx_compiler=clang++

[settings]
compiler=clang
compiler.version=8
compiler.libcxx=libc++
os=Android
os.api_level=16
arch=armv7
build_type=Release

[env]
CONAN_CMAKE_TOOLCHAIN_FILE=$ndk_root/build/cmake/android.toolchain.cmake
CONAN_CMAKE_FIND_ROOT_PATH=$standalone_toolchain/sysroot
PATH=[$standalone_toolchain/bin]
CHOST=$target_host
AR=\$target_host-ar
AS=$target_host-as
RANLIB=$target_host-ranlib
CC=$target_host-$cc_compiler
CXX=$target_host-$cxx_compiler
LD=$target_host-ld
STRIP=$target_host-strip
CFLAGS= -fPIE -fPIC -I$standalone_toolchain/include/c++/4.9.x
CXXFLAGS= -fPIE -fPIC -I$standalone_toolchain/include/c++/4.9.x
LDFLAGS= -pie

```

### PR DESCRIPTION
**Pull request by SSE4**
closes #28 #175 https://github.com/conan-community/community/issues/287

Specify library name and version:  **openssl/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
@lasote seems to be some kind of disk cleanup issue? various files are missing in various builds

**Comment by lasote** [comment]
@SSE4 thanks for reporting. Yes, I think we have an issue there.

**Comment by SSE4** [comment]
@lasote no hay problema! it's okay on these early stages I believe

**Comment by lasote** [comment]
We are rotating the apple-clang slaves to generate the apple-clang 11 packages, that's why this is locked. I'm going to launch a build for all packages for that config and then we can resume this one.

File: recipes/openssl/ALL/conanfile.py [hunk 1]
```diff
@@ -517,9 +521,28 @@ def build(self):
                     self._patch_makefile_org()
                 self._make()
 
+    @staticmethod
```
Comment (lasote) [inline recipes/openssl/ALL/conanfile.py]: Does all this code smell like something to fix in Conan? Something like `tools.cross_building(self.settings, native_x86=True)` or similar?

Comment (ericLemanissier) [reply recipes/openssl/ALL/conanfile.py]: As far as I am concerned, this is a missing feature in conan. I had to implement similar conditions several times already.

Comment (lasote) [reply recipes/openssl/ALL/conanfile.py]: Could you open an issue, please?

Comment (SSE4) [reply recipes/openssl/ALL/conanfile.py]: if we agree, I am going to open an issue in conan. I had to apply same in `icu`, for example

Comment (lasote) [reply recipes/openssl/ALL/conanfile.py]: For the future, if you are repeating the same workaround talk with us!

Comment (SSE4) [reply recipes/openssl/ALL/conanfile.py]: opened https://github.com/conan-io/conan/issues/5870

**Comment by SSE4** [comment]
ping @uilianries

**Pull request merged**